GUI/Types/Renderer/SelectedNodeRenderer.cs

```diff
@@ -229,6 +356,33 @@ public override void Render(Scene.RenderContext context)
 
             foreach (var node in selectedNodes)
             {
+                NodeClosestVertexInfo.TryGetValue(node, out var closestVertexInfo);
+                NodeVertexInfos.TryGetValue(node, out var vertexInfos);
+
+                if (vertexInfos == null)
+                {
+                    vertexInfos = GetAABBVertices(node.LocalBoundingBox, node.Transform);
+
+                    NodeVertexInfos.Add(node, vertexInfos);
+                    UpdateBuffer();
+                }
+
+                if (closestVertexInfo == null)
+                {
+                    closestVertexInfo = GetClosestVertexInfo(context.Camera, vertexInfos);
+
+                    NodeClosestVertexInfo.Add(node, closestVertexInfo);
+                    UpdateBuffer();
+                }
+
+                var newClosestInfo = GetClosestVertexInfo(context.Camera, vertexInfos);
+
+                if (closestVertexInfo != newClosestInfo)
+                {
+                    NodeClosestVertexInfo[node] = newClosestInfo;
+                    UpdateBuffer();
```

Comment:
This class already contains UpdateEveryFrame bool for particles, so with the dynamic drawing here, we probably always want to recalculate the buffer then.